### ISSUE TITLE
Training is cancelled when check_val_every_n_epoch > 1 and early stopping is used.

### ISSUE THREAD

**Issue by selflein** [PyTorchLightning/pytorch-lightning#6113]
## 🐛 Bug

Using early stopping callback on metric emitted during validation with `check_val_every_n_epoch` > 1 flag of trainer leads to the training being aborted.

Reproduced [here](https://colab.research.google.com/drive/1WimUPkXd7BPaBaErUDn4YvL0tKz6VSfg?usp=sharing
).

### Expected behavior

Training should continue even when `check_val_every_n_epoch` > 1 and early stopping is used. 
One has to think how the `patience` parameter  of the early stopping is handled in this case.

I see two possible directions

* Interpreting `patience` as the number of validation runs which did satisfy the early stopping criterion.
    This would mean e.g. choosing `check_val_every_n_epoch` = 2 and a `patience` = 10 would train for an additional 2 * 10 = 20 epochs which might not be what one would expect.

* Interpreting `patience` as the number of training epochs since the last validation improvement which have to be **at least** completed until the training is stopped, i.e., if the number of training epochs since the last improvement as measured by validation is larger than `patience` then stop. 

### Environment

```
* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     True
	- pyTorch_version:   1.7.0+cu101
	- pytorch-lightning: 1.2.0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020
```

**Comment by awaelchli** [comment]
Hi, you reported this with 1.2.0 but I'm struggling to get this failing. 
In terms of expected behaviour, we try to explain this in the docs of EarlyStopping:

> It must be noted that the patience parameter counts the number of validation checks with
>                 no improvement, and not the number of training epochs. Therefore, with parameters
>                 ``check_val_every_n_epoch=10`` and ``patience=3``, the trainer will perform at least 40 training
>                 epochs before being stopped.

Recently @ananthsub added the option `check_on_train_epoch_end`, which means now you can essentially simulate what you want: The validation runs on only on every N steps but the early stopping checks and counts internally with the training epoch using the last available value for the monitor.

**Comment by awaelchli** [comment]
Currently we show a message like this for stopping:

> Monitored metric val/loss did not improve in the last 3 epochs. Best score: 0.092. Signaling Trainer to stop.

We should change the wording "epochs" to something else. Suggestion?

**Comment by selflein** [comment]
Yes, just checked again and the error seems to be fixed with 1.2.10. 

I would vote for naming it something along the lines of "validation runs" which would be quite intuitive for me.

**Comment by awaelchli** [comment]
Yes I agree it would be intuitive, except we have now the setting in which we can let EarlyStopping run purely on the training epoch end (no validation), so then the wording would again be misleading 😅 
In the linked PR I currently have `f"Monitored metric {self.monitor} did not improve in the last {self.wait_count} records."`
